### PR TITLE
Improve service list api performance

### PR DESCRIPTION
**Pull request by jakolehm**
Calculates instance counts via aggregations.

### CODE REVIEW

File: server/app/views/v1/grid_services/index.json.jbuilder [hunk 1]
```diff
@@ -1,3 +1,32 @@
+total_counts = Container.collection.aggregate([
+  { :$match => { grid_id: @grid.id } },
+  { :$group => { _id: "$grid_service_id", total: {:$sum => 1} } }
+])
+running_counts =  Container.collection.aggregate([
+  { :$match => { :grid_id => @grid.id, :'state.running' => true } },
+  { :$group => { _id: "$grid_service_id", total: {:$sum => 1} } }
+])
+
```
Comment (jnummelin) [inline server/app/views/v1/grid_services/index.json.jbuilder]: Could the aggregations rather be part of model? For me it's always a bit weird to run DB operations on view level.

**Comment by jnummelin** [comment]
I'm good with subset of service json.

File: server/app/views/v1/grid_services/index.json.jbuilder (hunk 1 above)
Comment (jakolehm) [inline server/app/views/v1/grid_services/index.json.jbuilder]: You mean something like this:
```
total_counts = Container.counts_for_grid(@grid.id)
```
?

Comment (jnummelin) [inline server/app/views/v1/grid_services/index.json.jbuilder]: Yes, something along those lines

**Comment by jakolehm** [comment]
Optimized count lookups and changed json output back to full service json (too small perf gains).

File: server/app/views/v1/grid_services/_grid_service.json.jbuilder [hunk 2]
```diff
@@ -29,8 +29,13 @@ json.strategy grid_service.strategy
...
 json.pid grid_service.pid
 json.instances do
-  json.total grid_service.containers.count
-  json.running grid_service.containers.where('state.running' => true).count
+  if defined? instance_counts
+    json.total instance_counts[:total]
+    json.running instance_counts[:running]
+  else
+    json.total grid_service.containers.count
+    json.total grid_service.containers.count
```
Comment (jnummelin) [inline server/app/views/v1/grid_services/_grid_service.json.jbuilder]: Double total amount, I guess the other should be running count?

Comment (jakolehm) [inline server/app/views/v1/grid_services/_grid_service.json.jbuilder]: Good catch! Fixed.

**Comment by jnummelin** [comment]
LGTM after Mr. Travis also confirms :D

**Pull request merged**